addons/account_edi_ubl_cii/models/account_edi_format.py

```diff
@@ -174,6 +174,12 @@ def _create_invoice_from_xml_tree(self, filename, tree, journal=None):
         # EXTENDS account_edi
         self.ensure_one()
 
+        context_move_type = self._context.get('default_move_type', None)
+        if context_move_type:
+            last_move = self.env['account.move'].search([('move_type', '=', context_move_type)], limit=1)
```

Comment:
Hello @rigr-odoo :)
This can return `False` if there is no move with this type in the database and the code will fallback on the purchase journal. It will produce this error and not extract the data from the file in the case of `out_invoice`:
`Cannot create an invoice of type out_invoice with a journal having purchase as type.`

Reply:
Hi,
Thanks for the feedback ! :) 
This will be redone in another task to unify the uploading of invoices across the chatter, the upload buttons, and the Documents app.
I'm closing this PR for now.